src/lib/libeventloop.js

```diff
@@ -135,11 +135,17 @@ LibraryJSEventLoop = {
       {{{ runtimeKeepalivePop() }}}
       callUserCallback(() => {
         if ({{{ makeDynCall('idp', 'cb') }}}(t, userData)) {
+          {{{ runtimeKeepalivePush() }}}
+#if ENVIRONMENT_MAY_BE_NODE
+          var remaining = Math.max(0, n - _emscripten_get_now())
+#else
           // Save a little bit of code space: modern browsers should treat
           // negative setTimeout as timeout of 0
           // (https://stackoverflow.com/questions/8430966/is-calling-settimeout-with-a-negative-delay-ok)
-          {{{ runtimeKeepalivePush() }}}
-          setTimeout(tick, n - _emscripten_get_now());
+          // Node, however, gives TimeoutNegativeWarning
+          var remaining = n - _emscripten_get_now()
+#endif
+          setTimeout(tick, remaining);
```

Comment:
I can't write a code suggestion for the lines here (github complains some were deleted), but how about
```
var remaining = n - _emscripten_get_now();
#if ENVIRONMENT_MAY_BE_NODE
remaining = Math.max(0, remaining);
#endif

Reply:
sgtm